src/app/services/DataStore.ts

```diff
@@ -15,11 +15,13 @@ export class DataStore {
   allStories: Story[];
   userReservations: Reservation[];
   userReviews: Review[];
+  storyReviews: { string?: Review[] } = {};
 
   private _loggedInUserSubject = new Subject<User>();
   private _allStoriesSubject = new Subject<Story[]>();
   private _reservationsSubject = new Subject<Reservation[]>();
   private _reviewsSubject = new Subject<Review[]>();
+  private _storyReviewsSubjects: { string?: Subject<Review[]> } = {};
```

Comment:
I've never seen syntax like this before. I'm guessing it works, but I'd be curious to hear what it's doing some time in person.

Reply:
Oh yeah. I did that mostly because I forgot there was an easily used Map class haha. I changed it to a Map. I had actually never done anything like that before, but the idea was just that I wanted an object that contained specific classes, but that's really just a Map, so that makes more sense to use.